sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -481,11 +572,14 @@ public String getId() {
         ModelBridgeInternal.setPartitionKey(options, partitionKey);
         @SuppressWarnings("unchecked")
         Class<T> itemType = (Class<T>) item.getClass();
-        return this.getDatabase()
-                   .getDocClientWrapper()
-                   .replaceDocument(getItemLink(itemId), doc, ModelBridgeInternal.toRequestOptions(options))
-                   .map(response -> ModelBridgeInternal.createCosmosAsyncItemResponse(response, itemType))
-                   .single();
+        final CosmosItemRequestOptions requestOptions = options;
+        return withContext(context -> replaceItem(itemType, itemId, doc, requestOptions,context)).subscriberContext(reactorContext -> {
+            Optional<String> master = reactorContext.getOrEmpty(TracerProvider.MASTER_CALL);
+            if (master.isPresent()) {
+                reactorContext = reactorContext.put(TracerProvider.NESTED_CALL, true);
+            }
+            return reactorContext.put(TracerProvider.MASTER_CALL, true);
```

Comment:
can we factor out this code, here and other places?

e.g,
```java
return withContext(context -> replaceItem(itemType, itemId, doc, requestOptions,context))
    .subscriberContext(reactorContext -> factoredOutFunc())
```

Reply:
Done